internal/controller/vrg_kubeobjects.go

```diff
@@ -627,16 +602,60 @@ func (v *VRGInstance) getRecoverOrProtectRequest(
 		}
 }
 
+func (v *VRGInstance) getCaptureRequests() (map[string]kubeobjects.Request, error) {
+	captureRequestsStruct, err := v.reconciler.kubeObjects.ProtectRequestsGet(
+		v.ctx, v.reconciler.APIReader, v.veleroNamespaceName(), util.OwnerLabels(v.instance))
+	if err != nil {
+		return nil, fmt.Errorf("kube objects capture requests query error: %v", err)
+	}
+
+	return kubeobjects.RequestsMapKeyedByName(captureRequestsStruct), nil
+}
+
+func (v *VRGInstance) getRecoverRequests() (map[string]kubeobjects.Request, error) {
+	recoverRequestsStruct, err := v.reconciler.kubeObjects.RecoverRequestsGet(
+		v.ctx, v.reconciler.APIReader, v.veleroNamespaceName(), util.OwnerLabels(v.instance))
+	if err != nil {
+		return nil, fmt.Errorf("kube objects recover requests query error: %v", err)
+	}
+
+	return kubeobjects.RequestsMapKeyedByName(recoverRequestsStruct), nil
+}
+
 func (v *VRGInstance) kubeObjectsRecoveryStartOrResume(
-	result *ctrl.Result, s3StoreAccessor s3StoreAccessor,
-	sourceVrgNamespaceName, sourceVrgName string,
+	result *ctrl.Result, s3ProfileName string,
 	captureToRecoverFromIdentifier *ramen.KubeObjectsCaptureIdentifier,
-	captureRequests, recoverRequests map[string]kubeobjects.Request,
-	veleroNamespaceName string, labels map[string]string, log logr.Logger,
+	log logr.Logger,
 ) error {
+	veleroNamespaceName := v.veleroNamespaceName()
```

Comment:
If we are already refactoring - I think we can just use v.veleroNamespaceName() in the exact places we are calling to v.reconciler.kubeObjects.ProtectRequestCreate or v.reconciler.kubeObjects.ProtectRequestDelete or v.reconciler.kubeObjects.ProtectRequestGet. But instead we are passing it as a parameter everywhere.

Reply:
Done